File: src/tareas/ahorcado/ReproductorMidi.java
package tareas.ahorcado;

import java.io.BufferedInputStream;
import java.io.File;
import java.io.FileInputStream;
import java.io.InputStream;
import javax.sound.midi.MidiSystem;
import javax.sound.midi.Sequencer;

class ReproductorMidi {

    Sequencer sequencer;

    public ReproductorMidi(String ficheroMidi) throws Exception {
        //simplificamos el throws con Exception para hacer código limpio (aunque de peor calidad)
        // lo más fácil pillar el secuenciador por defecto.
        sequencer = MidiSystem.getSequencer();
        //antes de usar el secuenciaador hay que abrirlo.
        sequencer.open();
        // la secuencia midi es un InputStream. Puede ser un fichero, un stream de internet, ...en este caso es un fichero
        InputStream is = new BufferedInputStream(new FileInputStream(new File(ficheroMidi)));
        sequencer.setSequence(is);
        //cuando nos interesa arrancamos la reproducción.
        sequencer.start();
        //repetimos indefinidadmente la cancion indicando que se repita LOPP_CONTINUOSLY
        sequencer.setLoopCount(Sequencer.LOOP_CONTINUOUSLY);
    }

    void cerrar() {
        sequencer.close();
    }
}

File: src/tareas/ahorcado/ReproductorMidi.kt
package tareas.ahorcado

import java.io.BufferedInputStream
import java.io.File
import java.io.FileInputStream
import java.io.InputStream
import javax.sound.midi.MidiSystem
import javax.sound.midi.Sequencer


internal class ReproductorMidi(ficheroMidi: String) {
    //simplificamos el throws con Exception para hacer código limpio (aunque de peor calidad)
    // lo más fácil pillar el secuenciador por defecto.
    var sequencer: Sequencer = MidiSystem.getSequencer()

    init {
        //antes de usar el secuenciaador hay que abrirlo.
        sequencer.open()
        // la secuencia midi es un InputStream. Puede ser un fichero, un stream de internet, ...en este caso es un fichero
        val `is`: InputStream = BufferedInputStream(FileInputStream(File(ficheroMidi)))
        sequencer.setSequence(`is`)
        //cuando nos interesa arrancamos la reproducción.
        sequencer.start()
        //repetimos indefinidadmente la cancion indicando que se repita LOPP_CONTINUOSLY
        sequencer.loopCount = Sequencer.LOOP_CONTINUOUSLY
    }

    fun cerrar() {
        sequencer.close()
    }
}